allennlp/data/dataset_readers/huggingface_datasets_reader.py

```diff
@@ -0,0 +1,222 @@
+from typing import Iterable, Optional
+
+from allennlp.data import DatasetReader, Token, Field
+from allennlp.data.fields import TextField, LabelField, ListField
+from allennlp.data.instance import Instance
+from datasets import load_dataset, Dataset, DatasetDict
+from datasets.features import ClassLabel, Sequence, Translation, TranslationVariableLanguages
+from datasets.features import Value
+
+# TODO pab complete the documentation comments
+class HuggingfaceDatasetReader(DatasetReader):
+    """
+    This reader implementation wraps the huggingface datasets package
+    to utilize it's dataset management functionality and load the information in AllenNLP friendly formats
+    Note: Reader works w.r.t to only one split of the dataset,
+    i.e. you would need to create separate reader for separate splits
+
+    Following dataset and configurations have been verified and work with this reader
+
+            Dataset                       Dataset Configuration
+            `xnli`                        `ar`
+            `xnli`                        `en`
+            `xnli`                        `de`
+            `xnli`                        `all_languages`
+            `glue`                        `cola`
+            `glue`                        `mrpc`
+            `glue`                        `sst2`
+            `glue`                        `qqp`
+            `glue`                        `mnli`
+            `glue`                        `mnli_matched`
+            `universal_dependencies`      `en_lines`
+            `universal_dependencies`      `ko_kaist`
+            `universal_dependencies`      `af_afribooms`
+            `afrikaans_ner_corpus`        `NA`
+            `swahili`                     `NA`
+            `conll2003`                   `NA`
+            `dbpedia_14`                  `NA`
+            `trec`                        `NA`
+            `emotion`                     `NA`
+
+        # Parameters
+        dataset_name : `str`
+        config_name  : `str`, optional (default=`None`)
+        pre_load     : `bool`, optional (default='False`)
+    """
+
+    def __init__(
+        self,
+        max_instances: Optional[int] = None,
+        manual_distributed_sharding: bool = False,
+        manual_multiprocess_sharding: bool = False,
+        serialization_dir: Optional[str] = None,
+        dataset_name: str = None,
+        config_name: Optional[str] = None,
+        pre_load: Optional[bool] = False
+    ) -> None:
+        super().__init__(
+            max_instances,
+            manual_distributed_sharding=True,
+            manual_multiprocess_sharding=True,
+            serialization_dir,
+        )
```

Comment:
We usually do this like so:

```suggestion
        self,
        dataset_name: str = None,
        config_name: Optional[str] = None,
        pre_load: Optional[bool] = False,
        **kwargs,
    ) -> None:
        super().__init__(
            manual_distributed_sharding=True,
            manual_multiprocess_sharding=True,
            **kwargs,
        )
```

Reply:
Done.